pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -477,6 +481,37 @@ func (c *Controller) setupIndex() *AmbientIndexImpl {
 		}
 	})
 
+	c.configController.RegisterEventHandler(gvk.KubernetesGateway, func(oldCfg config.Config, newCfg config.Config, ev model.Event) {
+		if ev == model.EventUpdate || ev == model.EventAdd {
+			var newAddressCount int
+			var status *k8sbeta.GatewayStatus
+			var networkAddr networkAddress
+			if newStatus, ok := newCfg.Status.(*k8sbeta.GatewayStatus); ok {
+				status = newStatus
+				newAddressCount = len(newStatus.Addresses)
+			}
+			if newAddressCount > 0 {
+				idx.mu.Lock()
+				defer idx.mu.Unlock()
+				ip := status.Addresses[0].Value
```

Comment:
is this always an ip? or can it also be a hostname https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.GatewayAddress

Reply:
https://github.com/istio/istio/blob/8e5051d121a7caa1058531408535b02dd144d870/pilot/pkg/config/kube/gateway/conversion.go#L2089 and https://github.com/istio/istio/blob/8e5051d121a7caa1058531408535b02dd144d870/pilot/pkg/config/kube/gateway/deploymentcontroller.go#L157 were added in #46500 which should, at least for now, result in an IP address for waypoints. I think there could probably be logic added to ignore non-Waypoint gateways though.

Reply:
Yes we should ignore hostname for now

Reply:
I added code so that if we can't parse we log an error and move on instead of a panic. I did hit some panics during upgrades from istiod which wrote a hostname to this istiod where it expects an IP. In this case at least the behavior is OK since the kube gateway service update will write an IP and then we can process it happily in `ambientindex`.